plugin-build/src/main/kotlin/io/sentry/android/gradle/util/SentryAndroidSdkChecker.kt

```diff
@@ -0,0 +1,56 @@
+package io.sentry.android.gradle.util
+
+import java.util.LinkedList
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ResolvedDependency
+
+fun Project.getSentryAndroidSdkState(
+    configurationName: String,
+    variantName: String
+): SentryAndroidSdkState {
+    val configuration = configurations.findByName(configurationName)
+
+    if (configuration == null) {
+        logger.warn {
+            "Unable to find configuration $configurationName for variant $variantName."
+        }
+        return SentryAndroidSdkState.MISSING
+    }
+
+    val resolvedConfiguration = configuration.resolvedConfiguration
+    if (resolvedConfiguration.hasError()) {
+        resolvedConfiguration.rethrowFailure()
+        logger.warn { "Unable to resolve configuration $configurationName." }
+        return SentryAndroidSdkState.MISSING
+    }
+
+    val deps = resolvedConfiguration.firstLevelModuleDependencies
+    val version = deps.findSentryAndroidSdk()
+    if (version != null) {
+        return try {
+            val sdkState = SentryAndroidSdkState.from(version)
+            logger.info {
+                "Detected sentry-android $sdkState for version: $version, " +
+                    "variant: $variantName, config: $configurationName"
+            }
+            sdkState
+        } catch (e: IllegalStateException) {
+            logger.warn { e.localizedMessage }
```

Comment:
We are getting this when there's some garbage instead of a semver as version, e.g. `io.sentry:sentry-android:test` (which is probably not gonna happen anyway, because gradle would fail the sync), so I thought warning is a good middle-ground here, because error would mean we fail the build (which we don't), and info is not displayed by default in the output